nengo/processes.py

```diff
@@ -260,3 +260,129 @@ def step_presentinput(t):
+
+
+class Piecewise(Process):
+    """Present a piecewise input with different options for interpolation.
+
+    Given an input of tp = [0, 0.5, 0.75, 1], yp = [[0], [1], [-1], [0]] this will generate a
+    function that returns the values in yp at corresponding time points.
+
+    Elements in tp must be times (floats or ints). The values in yp can be floats for
+    1d or lists for multi-dimensional function. All lists must be of the same length.
+
+    Interpolations on the data points using scipy.interpolation are also supported. The default
+    interpolation is 'zero', which simply creates a piecewise function whose values begin
+    at the specified time points. So the above example would be shortcut for:
+        def function(t):
+            if t < 0.5:
+                return 0
+            elif t < 0.75
+                return 1
+            elif t < 1:
+                return -1
+            else:
+                return 0
+
+    For times before the first specified time, it will default to zero (of
+    the correct length). This means that the above can be simplified to:
+
+        Piecewise([0.5, 0.75, 1], yp = [[1], [-1], [0]])
+
+    Parameters
+    ----------
+    tp : time points of the function where values are specified
+    yp : values of the function at each time points, can be floats of lists
+    interpolation : optional parameter for interpolation. Can use 'linear', 'nearest', 'slinear',
+        'quadratic', 'cubic', 'zero'. The default is 'zero', which just creates a plain piecewise function whose
+        values begin at corresponding time points
+
+    Returns
+    -------
+    function:
+        Either a piecewise or interpolated function that takes a variable t and
+        returns the corresponding value,
+
+    Examples
+    --------
+
+      >>> func = piecewise({0.5: 1, 0.75: -1, 1: 0})
+      >>> func(0.2)
+      [0]
+      >>> func(0.58)
+      [1]
+
+      >>> func = piecewise({0.5: [1, 0], 0.75: [0, 1]})
+      >>> func(0.2)
+      [0,0]
+      >>> func(0.58)
+      [1,0]
+      >>> func(100)
+      [0,1]
+    """
+
+    data = PiecewiseData('data')
+    interpolation = EnumParam('interpolation', values=(
+        'zero', 'linear', 'nearest', 'slinear', 'quadratic', 'cubic'))
+
+    def __init__(self, data, interpolation='zero', **kwargs):
+        self.data = data
+        self.interpolation = interpolation
+
+        if self.interpolation != 'zero':
+            try:
+                import scipy.interpolate
+                self.sp_interpolate = scipy.interpolate
+            except ImportError:
+                raise ValidationError(
+                    "To interpolate, Scipy must be installed",
+                    attr='interpolation', obj=self)
+
+            if self._output_length() != 1:
+                raise ValidationError(
+                    "Interpolation is only supported for 1d output",
+                    attr='interpolation', obj=self)
+
+            if not self._interpolatable():
```

Comment:
I think I would warn and switch to no interpolation here rather than raising a validation error.